src/token_standards/xtcMethods.ts

```diff
@@ -61,14 +61,17 @@ const burnXTC = async (
   actor: ActorSubclass<BaseXtcService>,
   { to, amount }: BurnParams
 ): Promise<BurnResult> => {
-  const decimals = getDecimals(await getMetadata(actor));
+  const decimals = getDecimalsFromMetadata(await getMetadata(actor));
   const parsedAmount = parseAmountToSend(amount, decimals);
   return actor._burn({ canister_id: to, amount: parsedAmount });
 };
 
+const getDecimals = async (actor: ActorSubclass<BaseXtcService>) => getDecimalsFromMetadata(await getMetadata(actor))
```

Comment:
same and it seems it can be abstracted

Reply:
It cannot because it use getMetadata, We already had problems with that